homeassistant/components/launch_library/sensor.py

```diff
@@ -72,25 +76,55 @@ async def async_setup_entry(
 
     async_add_entities(
         [
-            NextLaunchSensor(coordinator, entry.entry_id, name),
+            NextLaunchSensor(
+                coordinator,
+                entry.entry_id,
+                SensorEntityDescription(
+                    key="next_launch",
+                    icon="mdi:rocket-launch",
+                    name=name,
+                ),
+            ),
         ]
     )
 
 
-class NextLaunchSensor(CoordinatorEntity, SensorEntity):
-    """Representation of the next launch sensor."""
+class LaunchLibraryEntity(CoordinatorEntity, SensorEntity):
+    """Base entity class for launch library sensors."""
 
     _attr_attribution = ATTRIBUTION
-    _attr_icon = "mdi:rocket-launch"
     _next_launch: Launch | None = None
 
     def __init__(
-        self, coordinator: DataUpdateCoordinator, entry_id: str, name: str
+        self,
+        coordinator: DataUpdateCoordinator,
+        entry_id: str,
+        description: SensorEntityDescription,
     ) -> None:
-        """Initialize a Launch Library entity."""
+        """Initialize a Launch Library base entity."""
         super().__init__(coordinator)
-        self._attr_name = name
-        self._attr_unique_id = f"{entry_id}_next_launch"
+        self._attr_name = description.name
+        self._attr_unique_id = f"{entry_id}_{description.key}"
+
+    @property
+    def available(self) -> bool:
+        """Return if the sensor is available."""
+        return super().available and self._next_launch is not None
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._next_launch = next((launch for launch in self.coordinator.data), None)
+        super()._handle_coordinator_update()
+
+    async def async_added_to_hass(self) -> None:
+        """When entity is added to hass."""
+        await super().async_added_to_hass()
+        self._handle_coordinator_update()
```

Comment:
This part should not move, as these are directly linked to the next_launch sensor

Reply:
Unless that is used for_all_ sensors it should not move

Reply:
At this point it will be used for all sensors, but down the line when the PR in the upstream library are merged I will add other sensors. I can move the code below to a separate base entity, that inherits from `LaunchLibraryEntity` and is used by the `NextLaunchSensor`, would that be a better solution?
```python
    @property
    def available(self) -> bool:
        """Return if the sensor is available."""
        return super().available and self._next_launch is not None

    @callback
    def _handle_coordinator_update(self) -> None:
        """Handle updated data from the coordinator."""
        self._next_launch = next((launch for launch in self.coordinator.data), None)
        super()._handle_coordinator_update()

    async def async_added_to_hass(self) -> None:
        """When entity is added to hass."""
        await super().async_added_to_hass()
        self._handle_coordinator_update()
```

Reply:
At this point there is only 1 sensor, so yeah.

It has nothing to do with entity descriptions which this PR is about so it should not be split/moved until needed in a PR that adds more.

Reply:
Sorry this is my mistake, I can see why this is confusing. I should not have call this PR `Use EntityDescription in launch_library` as my intention was this PR would also move things into a base entity to make everything easier in the next PR and not to make it to big. I will move this part of the PR to next PR when i add another sensor. This will also make it much clearer what my intentions are. 😃

Reply:
Still new at this, and not a pro in any sense. But appreciation it a lot that you take the time to answer my questions 😃

Reply:
Please see my latest commit, it moves it back.